### DISCUSSION TITLE
Can CSRF be disabled for a given security firewall?

### DISCUSSION THREAD

**Discussion by Bilge** [symfony/symfony#44059] [Q&A] [answered]
Supposing the same application serves both a web app and an API, it makes sense that we may wish to have automatic CSRF protection enabled for the web app but disable CSRF protection for the API. What is the best way to achieve this with Symfony?

**Comment by Bilge** [comment, marked as answer]
Since we are using `FOSRestBundle`, this can be achieved with `disable_csrf_role`. In general, it can be achieved with a form type extension, the same as `FOSRestBundle` uses internally: [DisableCSRFExtension](https://github.com/FriendsOfSymfony/FOSRestBundle/blob/cb75a41227ed5a9028fa3028c8225dda5c7b3086/Form/Extension/DisableCSRFExtension.php).

**Comment by jfsenechal** [comment]
You can also use custom Authenticator  [](https://symfony.com/doc/current/security/custom_authenticator.html)
